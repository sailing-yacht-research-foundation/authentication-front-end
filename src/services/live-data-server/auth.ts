import axios from "axios";
import { SYRF_SERVER } from "services/service-constants";
import syrfRequest from "utils/syrf-request";
import { v4 as uuidv4 } from 'uuid';

let uuid = localStorage.getItem('uuid');

if (uuid === null) {
    uuid = uuidv4();
    if (uuid)
        localStorage.setItem('uuid', uuid);
}

export const login = ({ email, password }) => {
    return axios.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/auth/login`, {
        username: email,
        password: password,
        devToken: SYRF_SERVER.DEV_TOKEN
    }).then(response => {
        return {
            success: true,
            token: response.data.token,
            user: response.data
        }
    }).catch(error => {
        return {
            success: false,
            error: error
        }
    })
}

export const anonymousLogin = () => {
    return axios.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/auth/anonymous-login`, {
        id: uuid,
        devToken: SYRF_SERVER.DEV_TOKEN
    }).then(response => {
        return {
            success: true,
            token: response.data?.token,
            data: response.data
        }
    }).catch(error => {
        return {
            success: false,
            error: error
        }
    })
}

export const validateToken = (token) => {
    return syrfRequest.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/auth/validate-token`, {
        token: token,
        renew: false
    }).then(response => {
        return {
            success: true,
            token: response.data.token,
            user: response.data
        }
    }).catch(error => {
        return {
            success: false,
            error: error
        }
    })
}


export const renewToken = (refreshToken) => {
    return syrfRequest.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/auth/renew-token`, {
        refresh_token: refreshToken
    })
    .then(response => {
        return {
            success: true,
            data: response.data
        }
    }).catch(error => {
        return {
            sucess: false,
            error: error
        }
    })
}

export const register = (data) => {
    return axios.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/users`, data)
    .then(response => {
        return {
            success: true,
            user: response.data.user
        }
    }).catch(error => {
        return {
            sucess: false,
            error: error
        }
    })
}

export const verifyEmail = () => {
    return syrfRequest.put(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/users/send-verify-email`)
    .then(response => {
        return {
            success: true,
        }
    }).catch(error => {
        return {
            sucess: false,
            error: error
        }
    })
}

export const sendForgotPassword = (email) => {
    return syrfRequest.put(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/users/forget-password`, {
        email: email
    })
    .then(response => {
        return {
            success: true,
        }
    }).catch(error => {
        return {
            sucess: false,
            error: error
        }
    })
}