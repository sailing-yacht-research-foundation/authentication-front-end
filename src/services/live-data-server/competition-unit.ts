import { SYRF_SERVER } from "services/service-constants";
import syrfService from 'utils/syrf-request';

export const getCompetitionUnitList = () => {
  return syrfService.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units`, {
  }).then(response => {
      return {
        success: true,
        data: response.data
      }
  }).catch(error => {
    return {
      success: false,
      error: error
    }
  });
}


export const getCompetitionUnitById = (id) => {
  return syrfService.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${id}`, {
  }).then(response => {
      return {
        success: true,
        data: response.data
      }
  }).catch(error => {
    return {
      success: false,
      error: error
    }
  });
};