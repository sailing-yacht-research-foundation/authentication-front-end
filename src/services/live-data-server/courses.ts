import { SYRF_SERVER } from 'services/service-constants';
import syrfService from 'utils/syrf-request';

export const list = () => {
    return syrfService.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/courses`).then(response => {
        return {
            success: true,
            data: response.data
        }
    }).catch(error => {
        return {
            success: false,
            error: error
        }
    });
}

export const create = (competitionUnitId, courseSequencedGeometries) => {
    return syrfService.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/courses`, {
        courseSequencedGeometries: courseSequencedGeometries,
        competitionUnitId: competitionUnitId
    }).then(response => {
        return {
            success: true,
            data: response.data
        }
    }).catch(error => {
        return {
            success: false,
            error: error
        }
    });
}

export const deleteCourse = (courseId) => {
    return syrfService.delete(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/courses/${courseId}`).then(response => {
        return {
            success: true,
            data: response.data
        }
    }).catch(error => {
        return {
            success: false,
            error: error
        }
    });
}

export const update = (competitionUnitId, courseId, courseSequencedGeometries) => {
    return syrfService.put(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/courses/${courseId}`, {
        courseSequencedGeometries: courseSequencedGeometries,
        competitionUnitId: competitionUnitId
    }).then(response => {
        return {
            success: true,
            data: response.data
        }
    }).catch(error => {
        return {
            success: false,
            error: error
        }
    });
}

export const updateCourseGeometry = (courseId, courseSequencedGeometries) => {
    console.log(courseSequencedGeometries);
    return syrfService.put(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/courses/${courseId}/sequenced`, courseSequencedGeometries).then(response => {
        return {
            success: true,
            data: response.data
        }
    }).catch(error => {
        return {
            success: false,
            error: error
        }
    });
}

export const getByCompetitionUnit = (competitionUnitId) => {
    return syrfService.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${competitionUnitId}/course`).then(response => {
        return {
            success: true,
            data: response.data
        }
    }).catch(error => {
        return {
            success: false,
            error: error
        }
    });
}

export const getById = (courseId) => {
    return syrfService.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/courses/${courseId}`)
        .then(response => {
            return {
                success: true,
                data: response.data
            }
        }).catch(error => {
            return {
                success: false,
                error: error
            }
        });
}