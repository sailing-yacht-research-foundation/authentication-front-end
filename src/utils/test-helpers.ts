export const defineWindowMatchMedia = () => {
    Object.defineProperty(window, "matchMedia", { // have to define here because the tests will fail
        writable: true,
        value: jest.fn().mockImplementation(query => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // Deprecated
            removeListener: jest.fn(), // Deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        }))
    });
}