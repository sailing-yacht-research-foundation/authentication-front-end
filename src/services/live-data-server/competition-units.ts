import moment from 'moment';
import { SYRF_SERVER } from 'services/service-constants';
import { formatServicePromiseResponse, parseKeyword } from 'utils/helpers';
import syrfRequest from 'utils/syrf-request';

export const search = (params) => {
    const query: any = {
        bool: {
            must: []
        }
    };

    query.bool.must.push({
        query_string: {
            query: parseKeyword(params.keyword),
        },

    });

    if (params.hasOwnProperty('from_date')
        && params.from_date !== ''
        && moment(params.from_date).isValid()) {
        query.bool.must.push({
            range: {
                "approx_start_time_ms": {
                    "gte": moment(params.from_date).set({ hour: 0, minute: 0, second: 0 }).unix() * 1000,
                }
            }
        });
    }

    if (params.hasOwnProperty('to_date')
        && params.to_date !== ''
        && moment(params.to_date).isValid()) {
        query.bool.must.push({
            range: {
                "approx_start_time_ms": {
                    "lt": moment(params.to_date).set({ hour: 23, minute: 59, second: 59 }).unix() * 1000,
                }
            }
        });
    }

    const searchParams: any = {
        query: query,
        sort: [
            "_score",
            {
                "approx_start_time_ms": {
                    "order": "desc"
                }
            },
        ]
    };

    searchParams._source = [
        "id", "source", "name", "approx_start_point", "start_country", "start_city", "start_year", "start_month",
        "approx_start_time_ms", "event_name", "event", "event_description", "isOpen", "allowRegistration", "status"
    ]; // only the fields we need
    searchParams.from = params.hasOwnProperty('page') ? ((Number(params.page) - 1) * Number(params?.size)) : 0;
    searchParams.size = params.size ?? 10;

    return formatServicePromiseResponse(syrfRequest.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/search`, searchParams))
}

export const create = (calendarEventId, data) => {
    return formatServicePromiseResponse(syrfRequest.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/calendar-events/${calendarEventId}/competition-units`, {
        ...data
    }))
}

export const getAllByCalendarEventId = (calendarEventId, page, size = 10) => {
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/calendar-events/${calendarEventId}/competition-units`, {
        params: {
            page: page,
            size: size,
        }
    }));
}

export const getAllCompetitionUnits = (page) => {
    const userId: any = localStorage.getItem('user_id');
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units${!!userId ? `?createdById_eq=${userId}` : ''}`, {
        params: {
            page: page
        }
    }))
}

export const get = (calendarEventId, competitionUnitId) => {
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/calendar-events/${calendarEventId}/competition-units/${competitionUnitId}`))
}

export const update = (calendarEventId, competitionUnitId, data) => {
    return formatServicePromiseResponse(syrfRequest.put(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/calendar-events/${calendarEventId}/competition-units/${competitionUnitId}`, {
        ...data
    }))
}

export const deleteCompetitionUnit = (calendarEventId, competitionUnitId) => {
    return formatServicePromiseResponse(syrfRequest.delete(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/calendar-events/${calendarEventId}/competition-units/${competitionUnitId}`))
}

export const getCompetitionUnitList = () => {
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units`))
}

export const getCompetitionUnitById = (id) => {
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${id}`));
};

export const searchScrapedRaceById = (id: string) => {
    const searchParams: any = {
        query: {
            match: {
                '_id': id
            }
        },
    };

    searchParams._source = ["id", "name", "approx_start_point", "start_country", "approx_start_time_ms", "url", "source", "isOpen", "allowRegistration"];

    return formatServicePromiseResponse(syrfRequest.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/search`, searchParams))
}

export const cloneCourse = (fromCompetitionUnitId, toCompetitionUnitId, newName) => {
    return formatServicePromiseResponse(syrfRequest.put(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${toCompetitionUnitId}/clone-course`, {
        cloneFromCompetitionId: fromCompetitionUnitId,
        newName: newName
    }))
}

export const getAllCompetitionUnitsByEventIdWithSort = (calendarEventId, page) => {
    const userId: any = localStorage.getItem('user_id');
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/calendar-events/${calendarEventId}/competition-units?sort=createdAt&srdir=-1${!!userId ? `&createdById_eq=${userId}` : ''}`, {
        params: {
            page: page
        }
    }))
}

export const getSimplifiedTracksByCompetitionUnit = (id: string) => {
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${id}/simplified-tracks`))
}

export const getLegsByCompetitionUnit = (id: string) => {
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${id}/legs`))
}

export const getTimeByCompetitionUnit = (id: string) => {
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${id}/time`))
}

export const getCourseByCompetitionUnit = (id: string) => {
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${id}/course`))
}

export const getSuggestion = (fieldName, word) => {
    const searchParams = {
        "suggest": {
            "autocomplete": {
                "prefix": word,
                "completion": {
                    "field": `${fieldName}.completion`,
                    "skip_duplicates": true,
                    "fuzzy": {
                        "fuzziness": 1
                    }
                }
            }
        }
    };

    return formatServicePromiseResponse(syrfRequest.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/search`, searchParams))
}

export const getRaceViewsCount = (competitionUnitId) => {
    return formatServicePromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${competitionUnitId}/viewers-count`));
}

export const stopRace = (competitionUnitId) => {
    return formatServicePromiseResponse(syrfRequest.put(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${competitionUnitId}/stop-race`));
}

export const startRace = (competitionUnitId) => {
    return formatServicePromiseResponse(syrfRequest.put(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/${competitionUnitId}/prepare-race`));
}

/**
 * Get live and happening soon races.
 * @param duration // in months
 * @param distance in mile
 * @param page 
 * @param size 
 * @param coordinate { lat, lon }
 * @returns 
 */
export const getLiveAndUpcomingRaces = (duration: number = 1, distance: number = 1000, page: number = 1, size: number = 10, coordinate) => {
    const query: any = {
        bool: {
            must: {},
            filter: [],
            should: [],
        },
    };

    if (coordinate) {
        query.bool.filter.push({
            "geo_distance": {
                "distance": `${distance}nmi`,
                "approx_start_point.coordinates": {
                    "lat": coordinate.lat,
                    "lon": coordinate.lon
                }
            }
        })
    }

    query.bool.must.bool = { // endtime is greater than now or starttime greater than now AND less than duration.
        should: [
            {
                "range": {
                    "approximateEndTime": {
                        "gte": "now"
                    }
                }
            },
            {
                "range": {
                    "approx_start_time_ms": {
                        "gte": moment().set({ hour: 0, minute: 0, second: 0 }).unix() * 1000,
                        "lt": moment().add(duration, "months").set({ hour: 23, minute: 59, second: 59 }).unix() * 1000,
                    }
                }
            }
        ]
    }

    const searchParams: any = {
        query: query,
        sort: [
            {
                "approx_start_time_ms": {
                    "order": "asc"
                }
            },
        ]
    };

    if (coordinate) {
        searchParams.sort.push({
            "_geo_distance": {
                "approx_start_point.coordinates": {
                    "lat": coordinate.lat,
                    "lon": coordinate.lon
                },
                "order": "asc",
            }
        })
    }

    searchParams._source = [
        "id", "source", "name", "approx_start_point", "start_country", "start_city", "start_year", "start_month",
        "approx_start_time_ms", "event_name", "event", "event_description", "isOpen", "allowRegistration", "status"
    ]; // only the fields we need
    searchParams.from = (page - 1) * size;
    searchParams.size = size;

    return formatServicePromiseResponse(syrfRequest.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/competition-units/search`, searchParams))
}