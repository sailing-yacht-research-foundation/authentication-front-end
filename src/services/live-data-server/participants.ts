import { SYRF_SERVER } from 'services/service-constants';
import { formatRequestPromiseResponse } from 'utils/helpers';
import syrfRequest from 'utils/syrf-request';

export const getAllByCalendarEventId = (calendarEventId, page) => {
    return formatRequestPromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/calendar-events/${calendarEventId}/participants`, {
        params: {
            page: page
        }
    }))
}

export const getAllByCalendarEventIdWithFilter = (calendarEventId, page, assignMode) => {
    let assign: any = null;
    if (assignMode === 'all') {
        assign = null;
    } else {
        assign = assignMode === 'assigned';
    }
    return formatRequestPromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/calendar-events/${calendarEventId}/participants${assign !== null ? `?assigned=${assign}` : ''}`, {
        params: {
            page: page
        }
    }))
}

export const get = (id) => {
    return formatRequestPromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/participants/${id}`))
}

export const create = (data) => {
    return formatRequestPromiseResponse(syrfRequest.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/participants`, {
        ...data
    }))
}

export const update = (id, data) => {
    return formatRequestPromiseResponse(syrfRequest.put(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/participants/${id}`, {
        ...data
    }))
}

export const deleteParticipant = (id) => {
    return formatRequestPromiseResponse(syrfRequest.delete(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/participants/${id}`))
}

export const getAllByVesselParticipantId = (vesselParticipantId) => {
    return formatRequestPromiseResponse(syrfRequest.get(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/vessel-participants/${vesselParticipantId}/participants`))
}

export const registerParticipantsToVesselParticipant = (vesselParticipantId, participants: any[]) => {
    return formatRequestPromiseResponse(syrfRequest.post(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/vessel-participants/${vesselParticipantId}/participants`, {
        participantIds: participants
    }))
}

export const unregisterParticipantFromVesselParticipant = (vesselParticipantId, participantId) => {
    return formatRequestPromiseResponse(syrfRequest.delete(`${SYRF_SERVER.API_URL}${SYRF_SERVER.API_VERSION}/vessel-participants/${vesselParticipantId}/participants/${participantId}`))
}