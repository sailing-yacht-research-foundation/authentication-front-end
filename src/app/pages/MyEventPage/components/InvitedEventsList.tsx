import React from 'react';
import styled from 'styled-components';
import { Table, Space, Spin, Tag, Button, Tooltip } from 'antd';
import { useTranslation } from 'react-i18next';
import Lottie from 'react-lottie';
import Invitation from '../assets/invitation.json'
import { translations } from 'locales/translations';
import { LottieMessage, LottieWrapper, TableWrapper } from 'app/components/SyrfGeneral';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { renderEmptyValue, renderTimezoneInUTCOffset, truncateName } from 'utils/helpers';
import { TIME_FORMAT } from 'utils/constants';
import { BiCheckCircle } from 'react-icons/bi';
import { MdRemoveCircle } from 'react-icons/md';
import { AcceptInvitationModal } from 'app/pages/MyEventPage/components/modals/AcceptInvitationModal';
import { getMyInvitedEvents } from 'services/live-data-server/participants';
import { RejectInviteRequestModal } from './modals/RejectInviteRequestModal';

const defaultOptions = {
    loop: true,
    autoplay: true,
    animationData: Invitation,
    rendererSettings: {
        preserveAspectRatio: 'xMidYMid slice'
    }
};

export const InvitedEventLists = (props) => {

    const { t } = useTranslation();

    const { reloadInvitationCount } = props;

    const translate = {
        status_open_regis: t(translations.my_event_list_page.status_openregistration),
        status_public: t(translations.my_event_list_page.status_publicevent),
        status_private: t(translations.my_event_list_page.status_privateevent),
        anyone_canregist: t(translations.tip.anyone_can_register_event_and_tracking),
        anyone_canview: t(translations.tip.anyone_can_search_view_event),
        only_owner_canview: t(translations.tip.only_owner_cansearch_view_event)
    }

    const columns = [
        {
            title: t(translations.general.name),
            dataIndex: 'name',
            key: 'name',
            render: (text, record) => {
                return <Tooltip title={record?.event.name}>
                    <Link to={`/events/${record?.event.id}`}>{truncateName(record?.event.name)}</Link>
                </Tooltip>;
            },
        },
        {
            title: 'Status',
            dataIndex: 'isOpen',
            key: 'isOpen',
            render: (isOpen, record) => {
                return (
                    <StatusContainer>
                        {record?.event.isOpen ? (
                            <Tooltip title={translate.anyone_canregist} color="blue">
                                <StyledTag >{translate.status_open_regis}</StyledTag>
                            </Tooltip>
                        ) : (
                        <Tooltip title={translate.only_owner_canview}>
                            <StyledTag >{translate.status_private}</StyledTag>
                        </Tooltip>
                        )}
                    </StatusContainer>
                );
            }
        },
        {
            title: t(translations.my_event_list_page.start_date),
            dataIndex: 'approximateStartTime',
            key: 'start_date',
            render: (value, record) => {
                if (record.event) {
                    return [moment(record.event.approximateStartTime).format(TIME_FORMAT.date_text_with_time), record.event.approximateStartTime_zone, renderTimezoneInUTCOffset(record?.event.approximateStartTime_zone)].filter(Boolean).join(' ')
                }

                return renderEmptyValue(null);
            }
        },
        {
            title: t(translations.my_event_list_page.invited_at),
            dataIndex: 'createdAt',
            key: 'createdAt',
            render: (value, record) => moment(record?.createdAt).format(TIME_FORMAT.date_text),
        },
        {
            title: 'Action',
            key: 'action',
            render: (text, record) => {
                return <Space size="middle">
                    <Button type="primary" onClick={() => acceptInviteRequest(record)} icon={<BiCheckCircle style={{ marginRight: '5px' }} />}>
                        {t(translations.group.accept)}
                    </Button>
                    <Button onClick={() => rejectInviteRequest(record)} icon={<MdRemoveCircle style={{ marginRight: '5px' }} />} danger>
                        {t(translations.group.reject)}
                    </Button>
                </Space>;
            }
        },
    ];

    const [pagination, setPagination] = React.useState<any>({
        page: 1,
        total: 0,
        rows: [],
        size: 10
    });

    const [showRejectConfirmModal, setShowRejectConfirmModal] = React.useState<boolean>(false);

    const [showAcceptModal, setShowAcceptModal] = React.useState<boolean>(false);

    const [request, setRequest] = React.useState<any>({});

    const [isLoading, setIsLoading] = React.useState<boolean>(false);

    const getInvitations = async (page, size) => {
        setIsLoading(true);
        const response = await getMyInvitedEvents(page, size);
        setIsLoading(false);

        if (response.success) {
            setPagination({
                ...pagination,
                page: page,
                total: response.data.count,
                rows: response.data.rows,
                size: response.data.size
            });
        }
    }

    React.useEffect(() => {
        getInvitations(1, 10);
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    const onPaginationChanged = (page, size) => {
        getInvitations(page, size);
    }

    const acceptInviteRequest = (request) => {
        setRequest(request);
        setShowAcceptModal(true);
    }

    const rejectInviteRequest = (request) => {
        setRequest(request);
        setShowRejectConfirmModal(true);
    }

    const reloadParent = () => {
        getInvitations(pagination.page, pagination.size);
        reloadInvitationCount();
    }

    return (
        <>
            <AcceptInvitationModal reloadParent={reloadParent} request={request} showModal={showAcceptModal} setShowModal={setShowAcceptModal} />
            <RejectInviteRequestModal reloadParent={reloadParent} showModal={showRejectConfirmModal} setShowModal={setShowRejectConfirmModal} request={request} />
            {pagination.rows.length > 0 ? (
                <Spin spinning={isLoading}>
                    <TableWrapper>
                        <Table
                            scroll={{ x: "max-content" }}
                            columns={columns}
                            dataSource={pagination.rows}
                            pagination={{
                                defaultPageSize: 10,
                                current: pagination.page,
                                total: pagination.total,
                                pageSize: pagination.size,
                                onChange: onPaginationChanged,
                            }}
                        />
                    </TableWrapper>
                </Spin>
            ) : (
                <LottieWrapper>
                    <Lottie options={defaultOptions} height={400} width={400} />
                    <LottieMessage>{t(translations.my_event_list_page.you_dont_have_any_invitation)}</LottieMessage>
                </LottieWrapper>
            )}
        </>
    );
};

const StatusContainer = styled.div`
    max-width: 250px;
`;

const StyledTag = styled(Tag)`
    margin-top: 4px;
    margin-bottom: 4px;
`;