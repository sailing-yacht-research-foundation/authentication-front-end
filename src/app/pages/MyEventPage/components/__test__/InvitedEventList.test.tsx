import MyProvider from 'app/components/Provider';
import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { render } from '@testing-library/react';
import { InvitedEventLists } from '../InvitedEventsList';
import * as InvitedEventListModule from '../InvitedEventsList';
import { defineWindowMatchMedia } from 'utils/test-helpers';
import * as ParticipantServiceModule from 'services/live-data-server/participants';

const shallowRenderer = createRenderer();

describe('InvitedEventList', () => {

    beforeAll(() => {
        defineWindowMatchMedia();
    });

    it('should render and match snapshot', () => {
        shallowRenderer.render(
            <MyProvider>
                <InvitedEventLists />
            </MyProvider>
        );
        const renderedOutput = shallowRenderer.getRenderOutput();
        expect(renderedOutput).toMatchSnapshot();
    });

    it("It should have been rendered with correct received props", () => {
        const invitedEventListComponentSpy = jest.spyOn(InvitedEventListModule, 'InvitedEventLists');

        render(
            <MyProvider>
                <InvitedEventLists reloadInvitationCount={10} />
            </MyProvider>
        );

        expect(invitedEventListComponentSpy).toHaveBeenCalledWith(expect.objectContaining({
            reloadInvitationCount: 10
        }), {});
    });

    it('should call getMyInvitedEvents on component render', () => {
        const getEventsSpy = jest.spyOn(ParticipantServiceModule, 'getMyInvitedEvents');

        render(
            <MyProvider>
                <InvitedEventLists reloadInvitationCount={10} />
            </MyProvider>
        );

        expect(getEventsSpy).toHaveBeenCalled();
    });
});
