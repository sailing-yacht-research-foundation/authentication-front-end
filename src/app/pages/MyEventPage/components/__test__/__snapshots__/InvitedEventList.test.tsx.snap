// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitedEventList should render and match snapshot 1`] = `
<Provider
  store={
    Object {
      "createReducer": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "injectedReducers": Object {},
      "injectedSagas": Object {},
      "replaceReducer": [Function],
      "runSaga": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <InvitedEventLists />
</Provider>
`;
